src/Plugin/AddPathPlugin.php

```diff
@@ -46,14 +46,14 @@ public function __construct(UriInterface $uri)
      */
     public function handleRequest(RequestInterface $request, callable $next, callable $first)
     {
-        $identifier = spl_object_hash((object) $first);
-
-        if (!array_key_exists($identifier, $this->alteredRequests)) {
-            $request = $request->withUri($request->getUri()
-                ->withPath($this->uri->getPath().$request->getUri()->getPath())
-            );
-            $this->alteredRequests[$identifier] = $identifier;
-        }
+         $prepend = $this->uri->getPath();
+         $path = $request->getUri()->getPath();
+
+         if (substr($path, 0, strlen($prepend)) !== $prepend) {
```

Comment:
can we add a comment here with the gist of our reasoning why we ended up chosing this solution over more complicated things?